src/gsy_e/models/area/__init__.py

```diff
@@ -240,17 +242,88 @@ def update_descendants_strategy_prices(self):
             log.exception("area.update_descendants_strategy_prices failed.")
             return
 
-    def cycle_coefficients_trading(self, current_time_slot: DateTime) -> None:
-        """Perform operations that should be executed on coefficients trading cycle."""
-        self._current_time_slot = current_time_slot
-
     def get_results_dict(self):
         """Calculate the results dict for the coefficients trading."""
         if self.strategy is not None:
-            return self.strategy.state.get_results_dict(self._current_time_slot)
+            return self.strategy.state.get_results_dict(self._current_market_time_slot)
         return {}
 
 
+class CoefficientArea(AreaBase):
+    """Area class for the coefficient matching mechanism."""
+    def __init__(self, name: str = None, children: List["CoefficientArea"] = None,
+                 uuid: str = None,
+                 strategy: SCMStrategy = None,
+                 config: SimulationConfig = None,
+                 grid_fee_percentage: float = None,
+                 grid_fee_constant: float = None,
+                 coefficient_percent: float = 0.0,
+                 market_maker_rate_eur: float = (
+                         ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE / 100.),
+                 feed_in_tariff_eur: float = GlobalConfig.FEED_IN_TARIFF / 100.,
+                 trade_rate: float = 0.0
+                 ):
+        # pylint: disable=too-many-arguments
+        super().__init__(name, children, uuid, strategy, config, grid_fee_percentage,
+                         grid_fee_constant)
+        self._coefficient_percent = coefficient_percent
+        self._market_maker_rate_eur = market_maker_rate_eur
+        self._feed_in_tariff_eur = feed_in_tariff_eur
+        self._trade_rate = trade_rate
+        self.past_market_time_slot = None
+
+    def activate_coefficients(self, current_time_slot: DateTime) -> None:
```

Comment:
I was confused by the method name. AFAIU, the `SCMStrategy.activate()` is not dealing with any coefficients, but rather collects the energy settings of the asset. 
I know, that there is now bidding strategy for SCM assets, but how about to be consistent, as we use the term "strategy" even in the SCM case and call this method: `activate_strategies` or alternatively: `activate_energy_parameters`?